doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+.. _coverage-plan:
+
+Coverage Plan
+=============
+
+.. note::
+  Work to implement the functional coverage described in this plan is on-going and the plan itself is not yet complete.
+
+.. todo::
+  Branch prediction hasn't yet been considered, this will add more coverage points and alter some others
+
+Introduction
+------------
+
+Ibex functional coverage is split into two major categories:
+
+* Architectural coverage - which is concerned with instructions being executed and exercising various features of the RISC-V architecture (e.g. PMP) and does not consider the details of how this execution occurs.
+* Microarchitectural coverage - which is concerned with the specifics of the RTL operation, ensuring interesting corner cases are seen along with various micro-architectural events (e.g. the different kinds of stall) and combinations of them.
+
+Architectural coverage is not Ibex specific and can be determined directly from a trace of executed instructions and is handled by RISC-V DV, details can be found in the `RISC-V DV documentation <https://htmlpreview.github.io/?https://github.com/google/riscv-dv/blob/master/docs/build/singlehtml/index.html#document-coverage_model>`_.
```

Comment:
Nit: Long sentence with chained "and"s. Maybe "... not Ibex specific. It can be..." ?